### PR TITLE
refactor: switch to factory

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/cdk/directives/platform/platform.directive.ts [hunk 1]
```diff
@@ -1,20 +1,12 @@
...
 import {TuiPlatform} from '@taiga-ui/cdk/types';
 
 @Directive({
     selector: '[tuiPlatform]',
-    providers: [
-        {
-            provide: TUI_PLATFORM,
-            deps: [TuiPlatformDirective],
-            useFactory: ({tuiPlatform}: TuiPlatformDirective) =>
-                tuiPlatform || tuiPlatformFactory(),
```
Comment (waterplea) [inline projects/cdk/directives/platform/platform.directive.ts:12]: Don't remove this for now, just change `tuiPlatformFactory()` to `inject(TUI_PLATFORM, { skipSelf: true })`

File: projects/cdk/directives/platform/platform.directive.ts [hunk 2]
```diff
@@ -1,20 +1,12 @@
...
-        },
-    ],
 })
 export class TuiPlatformDirective {
     @Input()
     @HostBinding('attr.data-platform')
-    tuiPlatform: TuiPlatform | '' = '';
+    tuiPlatform: TuiPlatform | '' = inject(TUI_PLATFORM);
```
Comment (waterplea) [inline projects/cdk/directives/platform/platform.directive.ts:11]: Please roll this change back

**Review by waterplea — APPROVED** [review]

File: projects/cdk/directives/platform/platform.directive.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/directives/platform/platform.directive.ts:12]: fixed

File: projects/cdk/directives/platform/platform.directive.ts (hunk 2 above)
Comment (splincode) [reply projects/cdk/directives/platform/platform.directive.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**